cassandra/src/test/scala/org/apache/zeppelin/cassandra/ParagraphParserTest.scala

```diff
@@ -496,73 +545,180 @@ class ParagraphParserTest extends FlatSpec
+  }
+
+  "Parser" should "fail parsing describe type" in {
+    val queries ="Describe Type toto"
 
     val ex = intercept[InterpreterException] {
       parser.parseAll(parser.queries, queries)
     }
-    ex.getMessage should be(s"Invalid syntax for DESCRIBE KEYSPACE. It should comply to the pattern: ${DESCRIBE_KEYSPACE_PATTERN.toString}")
+    ex.getMessage should be(s"Invalid syntax for DESCRIBE TYPE. It should comply to the patterns: " +
+      s"${DESCRIBE_TYPE_WITH_KEYSPACE_PATTERN.toString} or ${DESCRIBE_TYPE_PATTERN.toString}")
   }
 
-  "Parser" should "parse describe table" in {
-    val queries ="Describe TaBlE toto;"
+  "Parser" should "parse describe types" in {
+    val queries ="Describe types;"
 
     val parsed = parser.parseAll(parser.queries, queries)
 
-    parsed.get should be(List(DescribeTableCmd(None,"toto")))
+    parsed.get(0) shouldBe a [DescribeTypesCmd]
   }
 
-  "Parser" should "parse describe table with keyspace" in {
-    val queries ="Describe TaBlE ks.toto;"
+  "Parser" should "fail parsing describe types" in {
+    val queries ="Describe types"
+
+    val ex = intercept[InterpreterException] {
+      parser.parseAll(parser.queries, queries)
+    }
+    ex.getMessage should be(s"Invalid syntax for DESCRIBE TYPES. It should comply to the pattern: ${DESCRIBE_TYPES_PATTERN.toString}")
+  }
+
+  "Parser" should "parse describe function" in {
+    val queries ="Describe function toto;"
 
     val parsed = parser.parseAll(parser.queries, queries)
 
-    parsed.get should be(List(DescribeTableCmd(Some("ks"),"toto")))
+    parsed.get should be(List(DescribeFunctionCmd(None,"toto")))
   }
 
-  "Parser" should "fail parsing describe table" in {
-    val queries ="Describe TaBlE toto"
+  "Parser" should "parse describe function with keyspace" in {
+    val queries ="Describe function ks.toto;"
+
+    val parsed = parser.parseAll(parser.queries, queries)
+
+    parsed.get should be(List(DescribeFunctionCmd(Some("ks"),"toto")))
+  }
+
+  "Parser" should "fail parsing describe function" in {
+    val queries ="Describe function toto"
 
     val ex = intercept[InterpreterException] {
       parser.parseAll(parser.queries, queries)
     }
-    ex.getMessage should be(s"Invalid syntax for DESCRIBE TABLE. It should comply to the patterns: " +
-      s"${DESCRIBE_TABLE_WITH_KEYSPACE_PATTERN.toString} or ${DESCRIBE_TABLE_PATTERN.toString}")
+    ex.getMessage should be(s"Invalid syntax for DESCRIBE FUNCTION. It should comply to the patterns: " +
+      s"${DESCRIBE_FUNCTION_WITH_KEYSPACE_PATTERN.toString} or ${DESCRIBE_FUNCTION_PATTERN.toString}")
   }
 
-  "Parser" should "parse describe type" in {
-    val queries ="Describe Type toto;"
+  "Parser" should "parse describe functions" in {
+    val queries ="Describe functions;"
 
     val parsed = parser.parseAll(parser.queries, queries)
 
-    parsed.get should be(List(DescribeUDTCmd(None,"toto")))
+    parsed.get(0) shouldBe a [DescribeFunctionsCmd]
```

Comment:
In fact **DescribeFunctionsCmd** is just an Scala **object** with no property unlike other object having keyspace / name as properties. In this case comparing against [DescribeFunctionsCmd] makes sense